### PR TITLE
Terra-toolkit Nightwatch Testing Uplift

### PR DESCRIPTION
**Pull request by emilyrohrbough**
### Summary
Terra-toolkit has be moved it its own repository. Those changes allow cleanup the package level in regards to nightwatch testing. The changes in this PR include:
- Removes `terra-toolkit` from the repository 
- Removes `terra-toolkit` as a dev-dependency at the package level
- Updates nightwatch test scripts to use the root level `node_modules/terra-toolkit`
- Removes the `nightwatch.conf.js` file from each package

Addresses issue https://github.com/cerner/terra-core/issues/430 and https://github.com/cerner/terra-clinical/issues/82

### CODE REVIEW

File: packages/terra-arrange/CHANGELOG.md [hunk 1]
```diff
@@ -8,7 +8,8 @@ Unreleased
 ------------------
 ### Changed
 * Converted component to use CSS modules
-
+* Updated nightwatch test scripts
```
Comment (ryanthemanuel) [inline packages/terra-arrange/CHANGELOG.md]: All of these change logs should be in the "unreleased" section right?

File: packages/terra-modal/package.json [hunk 2]
```diff
@@ -48,9 +47,11 @@
     "test": "npm run test:spec && npm run test:nightwatch-default",
     "test:spec": "$(cd ..; npm bin)/jest --config ../../jestconfig.json",
     "test:all": "npm run test:nightwatch-default && npm run test:nightwatch-chrome && npm run test:nightwatch-firefox && npm run test:nightwatch-safari",
-    "test:nightwatch-default": "node ./node_modules/terra-toolkit/lib/scripts/nightwatch.js",
-    "test:nightwatch-chrome": "node ./node_modules/terra-toolkit/lib/scripts/nightwatch.js chrome",
-    "test:nightwatch-firefox": "node ./node_modules/terra-toolkit/lib/scripts/nightwatch.js firefox",
-    "test:nightwatch-safari": "node ./node_modules/terra-toolkit/lib/scripts/nightwatch-non-parallel.js safari"
+    "test:nightwatch-default": "node ../../node_modules/terra-toolkit/lib/scripts/nightwatch.js",
```
Comment (ryanthemanuel) [inline packages/terra-modal/package.json]: All of the commands should define the SPECTRE_TEST_SUITE environment variable

File: packages/terra-slide-group/package.json [hunk 3]
```diff
@@ -48,11 +47,11 @@
...
-    "test:remote": "WEBPACK_CONFIG_PATH=../../../../terra-site/webpack.config REMOTE=true SPECTRE_TEST_SUITE=terra-detail-view node ../../node_modules/terra-toolkit/lib/scripts/nightwatch-process.js --config tests/nightwatch.conf.js",
-    "test:remote:all": "WEBPACK_CONFIG_PATH=../../../../terra-site/webpack.config REMOTE=true SPECTRE_TEST_SUITE=terra-detail-view node ../../node_modules/terra-toolkit/lib/scripts/nightwatch-process.js --config tests/nightwatch.conf.js --env chrome-tiny,chrome-small,chrome-medium,chrome-large,chrome-huge,chrome-enormous,firefox-tiny,firefox-small,firefox-medium,firefox-large,firefox-huge,firefox-enormous,ie10-tiny,ie10-small,ie10-medium,ie10-large,ie10-huge,ie10-enormous,ie11-tiny,ie11-small,ie11-medium,ie11-large,ie11-huge,ie11-enormous,edge-tiny,edge-small,edge-medium,edge-large,edge-huge,edge-enormous,safari-tiny,safari-small,safari-medium,safari-large,safari-huge,safari-enormous"
+    "test:nightwatch-default": "SPECTRE_TEST_SUITE=terra-detail-view node ../../node_modules/terra-toolkit/lib/scripts/nightwatch.js",
```
Comment (ryanthemanuel) [inline packages/terra-slide-group/package.json]: This should be terra-slide-group

File: packages/terra-arrange/CHANGELOG.md (hunk 1 above)
Comment (bjankord) [inline packages/terra-arrange/CHANGELOG.md]: This is slated for release today. We can leave these here and update if we need to release tomorrow.

Comment (ryanthemanuel) [inline packages/terra-arrange/CHANGELOG.md]: OK cool.  Oh yeah just noticed the date :)

File: packages/terra-overlay/package.json [hunk 4]
```diff
@@ -22,8 +22,7 @@
   "homepage": "https://github.com/cerner/terra-core#readme",
   "devDependencies": {
     "react-intl": "^2.3.0",
-    "terra-props-table": "^0.8.0",
-    "terra-toolkit": "^1.0.0"
+    "terra-props-table": "^0.8.0"
```
Comment (bjankord) [inline packages/terra-overlay/package.json]: We should bump this to ^1.0.0

**Comment by mjhenkes** [comment]
Probably should update the readme.md too. to remove terra toolkit from the list of packages

File: packages/terra-overlay/package.json (hunk 4 above)
Comment (Matt-Butler) [inline packages/terra-overlay/package.json]: ^1.0.0

**Comment by bjankord** [comment]
See build errors along the following:

```bash
ERROR in ./packages/terra-i18n/lib/translationLoaders.js
Module not found: Error: Can't resolve 'en.js' in '/home/travis/build/cerner/terra-core/packages/terra-i18n/lib'
```

Gonna restart the build

File: packages/terra-app-delegate/package.json [hunk 5]
```diff
@@ -35,11 +32,11 @@
...
-    "test:remote": "WEBPACK_CONFIG_PATH=../../../../terra-site/webpack.config SPECTRE_TEST_SUITE=terra-app-delegate REMOTE=true node ../../node_modules/terra-toolkit/lib/scripts/nightwatch-process.js --config tests/nightwatch.conf.js",
-    "test:remote:all": "WEBPACK_CONFIG_PATH=../../../../terra-site/webpack.config SPECTRE_TEST_SUITE=terra-app-delegate REMOTE=true node ../../node_modules/terra-toolkit/lib/scripts/nightwatch-process.js --config tests/nightwatch.conf.js --env chrome-tiny,chrome-small,chrome-medium,chrome-large,chrome-huge,chrome-enormous,firefox-tiny,firefox-small,firefox-medium,firefox-large,firefox-huge,firefox-enormous,ie10-tiny,ie10-small,ie10-medium,ie10-large,ie10-huge,ie10-enormous,ie11-tiny,ie11-small,ie11-medium,ie11-large,ie11-huge,ie11-enormous,edge-tiny,edge-small,edge-medium,edge-large,edge-huge,edge-enormous,safari-tiny,safari-small,safari-medium,safari-large,safari-huge,safari-enormous"
+    "test:nightwatch-default": "SPECTRE_TEST_SUITE=terra-app-delegate node ../../node_modules/terra-toolkit/lib/scripts/nightwatch.js",
```
Comment (mhemesath) [inline packages/terra-app-delegate/package.json]: I think what would really clean this up is if we could make terra-toolit be a global module much like babel. Example:

```
test:terra-toolkit: terratoolkit -- config ../../terratoolkit.json
```
A few benefits to this. 
1. cleans up this repetitive config in packag.json
2. Decouples the tests from nightwatch and they are more of an implementation detail of terra toolkit.
3 allows us to look for a specific config file and potentially even search parent directories so we don't have redundant config. This config could also store `SPECTRE_TEST_SUITE` or derive it from the package.json

See: http://ourcodeworld.com/articles/read/393/how-to-create-a-global-module-for-node-js-properly

Comment (bjankord) [inline packages/terra-app-delegate/package.json]: Let's log an issue on the https://github.com/cerner/terra-toolkit for this work. We can tackle it in a separate PR. The scope of this PR is for moving terra-toolkit out of terra-core into it's own repo.

File: packages/terra-badge/package.json [hunk 6]
```diff
@@ -31,10 +31,12 @@
...
+    "test:nightwatch-safari": "SPECTRE_TEST_SUITE=terra-badge node ../../node_modules/terra-toolkit/lib/scripts/nightwatch-non-parallel.js safari",
+    "test:remote": "SPECTRE_TEST_SUITE=terra-badge REMOTE=true node ../../node_modules/terra-toolkit/lib/scripts/nightwatch-process.js --config ../../node_modules/terra-toolkit/lib/nightwatch.conf.js --env default-tiny,default-small,default-medium,default-large,default-huge,default-enormous",
+    "test:remote:all": "SPECTRE_TEST_SUITE=terra-badge REMOTE=true node ../../node_modules/terra-toolkit/lib/scripts/nightwatch-process.js --config ../../node_modules/terra-toolkit/lib/nightwatch.conf.js --env chrome-tiny,chrome-small,chrome-medium,chrome-large,chrome-huge,chrome-enormous,firefox-tiny,firefox-small,firefox-medium,firefox-large,firefox-huge,firefox-enormous,ie10-tiny,ie10-small,ie10-medium,ie10-large,ie10-huge,ie10-enormous,ie11-tiny,ie11-small,ie11-medium,ie11-large,ie11-huge,ie11-enormous,edge-tiny,edge-small,edge-medium,edge-large,edge-huge,edge-enormous,safari-tiny,safari-small,safari-medium,safari-large,safari-huge,safari-enormous"
```
Comment (bjankord) [inline packages/terra-badge/package.json]: We may want to remove the remote tests for the time being. Right now sauce labs throttles our usage so we never actually run the remote tests.

**Comment by bjankord** [comment]
@emilyrohrbough toolkit@1.2.2 is released which should help with translation path issues we're seeing on travisCI

**Pull request merged**